### ISSUE TITLE
How to Save Results as .nii.gz Instead of PNG?

### ISSUE THREAD

**Issue by Ziyan-Huang** [BAAI-DCAI/SegVol#31]
Great work on this project! I’ve been using it, and it’s working really well. However, I encountered a small issue when running the inference_demo.sh script. By default, it saves the results as PNG images, but I would like to save the segmentation masks in the same size as the original images, in .nii.gz format (NIfTI format).

Could you please guide me on how to modify the script to save the results as .nii.gz files?

I also noticed that your web demo seems to support this feature, but I’m hoping to run this locally. Any help would be appreciated!

Thank you!

**Comment by inhaowu** [comment]
I also have problems on that. Do you solve the problem?

**Comment by 2384441037** [comment]
def save_preds(ct_path, save_path, logits_mask, start_coord, end_coord):
        ct = nib.load(ct_path)
        logits_mask = logits_mask.transpose(-1, -3)
        start_coord[-1], start_coord[-3] = start_coord[-3], start_coord[-1]
        end_coord[-1], end_coord[-3] = end_coord[-3], end_coord[-1]
        preds_save = torch.zeros(ct.shape)
        
        print(f'start_coord:{start_coord}')
        print(f'end_coord:{end_coord}')
        print(f'ct.shape:{ct.shape}')
        print(f'logits_mask.shape:{logits_mask.shape}')
        print(f'preds_save.shape:{preds_save.shape}')
        
        target_shape = (
        end_coord[0] - start_coord[0],
        end_coord[1] - start_coord[1],
        end_coord[2] - start_coord[2],
    )
        if logits_mask.shape != target_shape:
            print(f"logits_mask.shape {logits_mask.shape} != target {target_shape}")
            start_coord[-1], start_coord[-3] = start_coord[-3], start_coord[-1]
            end_coord[-1], end_coord[-3] = end_coord[-3], end_coord[-1]
        preds_save[start_coord[0]:end_coord[0], 
                        start_coord[1]:end_coord[1], 
                        start_coord[2]:end_coord[2]] = torch.sigmoid(logits_mask)
        
        preds_save = torch.where(preds_save > 0.5, 1., 0.).numpy()
        preds_nii = nib.Nifti1Image(preds_save, affine=ct.affine, header=ct.header)
        nib.save(preds_nii, save_path)
        print(f'-----------------{save_path}保存成功-----------------------')

可以试一下这个
